### ISSUE TITLE
Slow "Mono llvmfullaot Pri0 Runtime Tests Run Linux arm64 release"

### ISSUE THREAD

**Issue by EgorBo** [dotnet/runtime#65626]
`Mono llvmfullaot Pri0 Runtime Tests Run Linux arm64 release` takes around 2.5H to finish.

There are some interesting anomalies in the logs, e.g.:
![image](https://user-images.githubusercontent.com/523221/154822948-ea049c01-fc25-4694-a412-5612fe30354e.png)
(I checked various runs)

It says that prejitting of a single managed assembly `Microsoft.Win32.SystemEvents.dll` takes almost **10** minutes 😮 (mostly in LLVM's opt+llc)

I parsed the output into an excel table:
![image](https://user-images.githubusercontent.com/523221/154823135-00d9df01-b948-436b-aaf6-3a13ff4b4f94.png)

is it possible to move some libs/tests to the outerloop? e.g. `JIT/Methodical/MDArray/GaussJordan/classarr_cs_do/classarr_cs_do` test. And I guess we need to figure out what exactly makes `Microsoft.Win32.SystemEvents.dll` so long to prejit - there are not much stuff in it.

cc @akoeplinger @vargaz @steveisok

**Comment by steveisok** [comment]
Adding @SamMonoRT

**Comment by vargaz** [comment]
Yes, these are very slow, they run opt+llc on unlinked assemblies.

**Comment by agocke** [comment]
This test run is pretty regularly timing out -- can we get someone to investigate if the slowness is a bug, or if we need to adjust the timeout?

**Comment by SamMonoRT** [comment]
> This test run is pretty regularly timing out -- can we get someone to investigate if the slowness is a bug, or if we need to adjust the timeout?

This PR should help ease the timeouts seen in last couple weeks. Even with that fix, the lane is 2.5+hrs long. Still discussing this, but we might possibly 1. want to exclude certain long running tests as part of PR runs in this lane, 2. Extend the timeout to stabilize CI in the short term

**Comment by agocke** [comment]
@SamMonoRT which PR?

**Comment by agocke** [comment]
Looks like that resolved the problem. I'm going to close this out for now.

**Comment by EgorBo** [comment]
It doesn't look fixed to me, every time this job is triggered it takes 4-5 hours, e.g. https://dev.azure.com/dnceng-public/public/_build/results?buildId=146722&view=results

**Comment by EgorBo** [comment]
I've wrote a quick parser for the output (for today's PR ^) and sorted assemblies by time it takes to run LLVM (opt and llc):

![image](https://user-images.githubusercontent.com/523221/214411207-eb05fd1a-262c-451a-bf91-9219bcca1872.png)

**Comment by EgorBo** [comment]
E.g. just by moving AdvSimd tests alone to an outerloop pipeline we can save ~30 minutes (4 dlls)

**Comment by steveisok** [comment]
> E.g. just by moving AdvSimd tests alone to an outerloop pipeline we can save ~30 minutes (4 dlls)

I think I'd rather move the whole thing out and then analyze what we can run per PR.

**Comment by steveisok** [comment]
@EgorBo thanks for putting together the updated list!

**Comment by SingleAccretion** [comment]
Wanted to mention as we should be careful to leave enough testing on PRs to reliably catch failures introduced by adding new Jit tests. In my experience these are not uncommon.

**Comment by SamMonoRT** [comment]
@kotlarmilos @vitek-karas - not sure if this is something your team owns now and what more remains here? Please can you re-assign as appropriate.

**Comment by steveisok** [comment]
I'll take this as it likely has to do w/ the aot compiler performance itself.

**Comment by agocke** [comment]
My general philosophy is, "PR is for fast reliable tests" so I agree with the theory of moving everything out, then moving things back in that meet that criteria. Ideally we can find the sweet spot of fast + high confidence in finding bugs.
